include/deal.II/lac/sparsity_pattern.h

```diff
@@ -1430,8 +1430,13 @@ SparsityPattern::load (Archive &ar, const unsigned int)
 
   ar &max_dim &rows &cols &max_vec_len &max_row_length &compressed &store_diagonal_first_in_row;
 
-  rowstart = new std::size_t [max_dim + 1];
-  colnums = new size_type [max_vec_len];
+  if (rowstart != 0)
+    delete[] rowstart;
+  rowstart = new std::size_t[max_dim + 1];
+
+  if (colnums != 0)
+    delete[] colnums;
+  colnums = new size_type[max_vec_len];
```

Comment:
I can never remember whether or not its legal. I'm probably to blame for a fair share of these `if`s. But then, they don't hurt either.